iot/api-client/http_example/src/main/java/com/google/cloud/iot/examples/HttpExample.java

```diff
@@ -0,0 +1,139 @@
+      // Could not parse the flags.
+      System.exit(1);
+    }
+
+    // Build the resource path of the device that is going to be authenticated.
+    String devicePath =
+        String.format(
+            "projects/%s/locations/%s/registries/%s/devices/%s",
+            options.projectId, options.cloudRegion, options.registryId, options.deviceId);
+
+    // This describes the operation that is going to be perform with the device.
+    String urlSuffix = options.messageType.equals("event") ? "publishEvent" : "setState";
+
+    String urlPath =
+        String.format(
+            "%s/%s/%s:%s", options.httpBridgeAddress, options.apiVersion, devicePath, urlSuffix);
+    URL url = new URL(urlPath);
+    System.out.format("Using URL: '%s'\n", urlPath);
+
+    // Create the corresponding JWT depending on the selected algorithm.
+    String token;
+    if (options.algorithm.equals("RS256")) {
+      token = createJwtRsa(options.projectId, options.privateKeyFile);
+    } else if (options.algorithm.equals("ES256")) {
+      token = createJwtEs(options.projectId, options.privateKeyFile);
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid algorithm " + options.algorithm + ". Should be one of 'RS256' or 'ES256'.");
+    }
+
+    // Data sent through the wire has to be base64 encoded.
+    Base64.Encoder encoder = Base64.getEncoder();
+
+    // Publish numMessages messages to the HTTP bridge.
+    for (int i = 1; i <= options.numMessages; ++i) {
+      String payload = String.format("%s/%s-payload-%d", options.registryId, options.deviceId, i);
+      System.out.format(
+          "Publishing %s message %d/%d: '%s'\n",
+          options.messageType, i, options.numMessages, payload);
+      String encPayload = encoder.encodeToString(payload.getBytes("UTF-8"));
+
+      HttpURLConnection httpCon = (HttpURLConnection) url.openConnection();
+      httpCon.setDoOutput(true);
+      httpCon.setRequestMethod("POST");
+
+      // Adding headers.
+      httpCon.setRequestProperty("Authorization", String.format("Bearer %s", token));
+      httpCon.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+
+      // Adding the post data. The structure of the data send depends on whether it is event or a
+      // state message.
+      JSONObject data = new JSONObject();
+      if (options.messageType.equals("event")) {
+        data.put("binary_data", encPayload);
+      } else {
+        JSONObject state = new JSONObject();
+        state.put("binary_data", encPayload);
+        data.put("state", state);
+      }
+      httpCon.getOutputStream().write(data.toString().getBytes("UTF-8"));
+      httpCon.getOutputStream().close();
+
+      // This will perform the connection as well.
+      System.out.println(httpCon.getResponseCode());
+      System.out.println(httpCon.getResponseMessage());
+
+      // Send events every second; states, every 5.
+      Thread.sleep(options.messageType.equals("event") ? 1000 : 5000);
```

Comment:
it might be worth pulling out that param to sleep in a separate statement. so that the sleep statement is clearer.